### DISCUSSION TITLE
Load extentions from DBeaver

### DISCUSSION THREAD

**Discussion by davealexis** [duckdb/duckdb#7106] [Q&A] [answered]
Is it possible to load DuckDB extentions in DBeaver? If so, how is it done?

**Comment by Mause** [comment]
Have a look at the docs for installing and loading extensions from SQL: https://duckdb.org/docs/extensions/overview

**Comment by davealexis** [comment]
I did. The INSTALL and LOAD (and other commands) work beautifully from the DuckDB CLI, but not in DBeaver, which says the "INSTALL" and "LOAD" are not valid. I was thinking that maybe there was a different syntax for doing these things from DBeaver or from the Java client that DBeaver uses? I could not find any documentation mentioning a different syntax, so I figured I'd ask here before I go to the DBeaver folks.

**Reply by Mause** [reply]
Can you give some more information about the error you're experiencing when you try to use those statements?

**Reply by davealexis** [reply]
So, let's take the sqlite_scanner extension (https://duckdb.org/docs/extensions/sqlite_scanner). 
In an active DuckDB SQL editor window in DBeaver, I run `INSTALL sqlite;` and I get  `SQL Error: Parser Error: syntax error at or near "install"`.
If I try something like `LOAD sqlite` I get `SQL Error: Parser Error: syntax error at or near "sqlite"`.
I tried downloading extensions from the extensions.duckdb.org site and try to load them with, e.g. `load 'c:\temp\sqlite_scanner.duckdb_extension.gz'` and I get `SQL Error: Invalid Input Error: File c:\temp\sqlite_scanner.duckdb_extension.gz could not be loaded`. 
It looks like it knows about the LOAD command

**Reply by davealexis** [reply, marked as answer]
Argh. Everything I see on the DuckDB site so far has been just awesome, so I got to thinking it has to work as described. So did some digging in DBeaver. It had version 0.3.1 of the DuckDB driver installed!!
Updated it to 0.7.1, and now it's just absolutely gorgeous! Load works. Install works. I'm in heaven!
